src/classes/BDI_MigrationMappingHelper.cls

```diff
@@ -224,10 +224,14 @@ public class BDI_MigrationMappingHelper {
 
         for (Data_Import_Field_Mapping__mdt reference : referenceFieldMappings) {
             if (UTIL_Namespace.shouldAlignNamespace) {
+                reference.Target_Object_Mapping__r.Object_API_Name__c = UTIL_Namespace.alignSchemaNSWithEnvironment(
+                        reference.Target_Object_Mapping__r.Object_API_Name__c);
+                reference.Target_Object_Mapping__r.Legacy_Data_Import_Object_Name__c = UTIL_Namespace.alignSchemaNSWithEnvironment(
```

Comment:
I think we can remove this line. The `Legacy_Data_Import_Object_Name__c` are an artifact from help text mappings and shouldn't ever have a namespace even in namespaced orgs. The only values we expect to have in there at this point are all in our packaged/protected object mapping metadata records. This line will probably never actually add/remove a namespace for these strings.